### PR TITLE
Sync with upstream

### PR DESCRIPTION
**Pull request by DeviaVir**
We reverted the sync with upstream last time, due to an issue unrelated to their updates. We're a few important bug fixes behind so getting our version back in sync is important to do every so often.

Reviewers: @aheckmann-pebble @jlai

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+sudo: required
```
Comment (thepwagner) [inline .travis.yml:1]: Can we flick the switch to use this too? Free and gives me warm/fuzzy feelings.

File: Makefile [hunk 2]
```diff
@@ -2,12 +2,12 @@
 .PHONY : test
 
 update-dependencies:
-	docker pull jwilder/docker-gen:latest
-	docker pull nginx:latest
+	docker pull jwilder/docker-gen:0.7.3
+	docker pull nginx:1.9.12
```
Comment (thepwagner) [inline Makefile:6]: Expected: `1.9.15`

**Comment by thepwagner** [comment]
+1

File: Makefile (hunk 2 above)
Comment (DeviaVir) [inline Makefile:6]: @lrvick voted to use `latest` for this, not sure if we want to stick to what upstream has or go rogue? (funny enough upstream just switched away from using `:latest` ^^^)

File: .travis.yml (hunk 1 above)
Comment (DeviaVir) [inline .travis.yml]: Done, but won't be active until this is merged, no `.travis.yml` before this

File: Makefile (hunk 2 above)
Comment (aheckmann-pebble) [inline Makefile:6]: IMO, latest requires testing each built image before deploy and not many of our apps do so (yet).

**Comment by aheckmann-pebble** [comment]
:+1:

File: .travis.yml (hunk 1 above)
Comment (DeviaVir) [inline .travis.yml:1]: Have no fear, travis is here 
screen shot 2016-07-14 at 18 45 35

**Pull request merged**